install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml

```diff
@@ -318,11 +318,18 @@ spec:
       - name: cilium-monitor
         image: {{ include "cilium.image" .Values.image | quote }}
         imagePullPolicy: {{ .Values.image.pullPolicy }}
-        command: ["cilium"]
+        command:
+        - /bin/bash
+        - -c
+        - --
         args:
-        - monitor
-        {{- range $type := .Values.monitor.eventTypes }}
-        - --type={{ $type }}
+        - |-
```

Comment:
Have you tested this? Looking at the syntax, I'm not sure the `\` at the end of the lines are valid. 

The resulting string will have a \ before a newline, which I don't think is what we want:
```json
{
    "args": [
      "for i in {1..5}; do \\\n  [ -S /var/run/cilium/monitor1_2.sock ] && break || sleep 10;\\\ndone; \\\ncilium monitor --type=trace --type=trace"
    ],
}
```

Reply:
it should work
i'll re-check

Reply:
kubectl describe
```
  cilium-monitor:
    Container ID:  containerd://cdf0e13ab5c14bebd10b465516cb2c1e02d25e8f246b052848f56a5fb0d4c58f
    Image:         quay.io/cilium/cilium:v1.11.3@sha256:cb6aac121e348abd61a692c435a90a6e2ad3f25baa9915346be7b333de8a767f
    Image ID:      quay.io/cilium/cilium@sha256:cb6aac121e348abd61a692c435a90a6e2ad3f25baa9915346be7b333de8a767f
    Port:          <none>
    Host Port:     <none>
    Command:
      /bin/bash
      -c
      --
    Args:
      for i in {1..5}; do \
        [ -S /var/run/cilium/monitor1_2.sock ] && break || sleep 10;\
        done; \
      cilium monitor --type=trace --type=drop
    State:          Running
      Started:      Thu, 14 Apr 2022 11:05:17 +0300
    Ready:          False
    Restart Count:  0
    Limits:
      memory:  128Mi
    Requests:
      cpu:        10m
      memory:     64Mi
    Startup:      http-get http://127.0.0.1:9876/healthz delay=0s timeout=1s period=2s #success=1 #failure=105
    Environment:  <none>
    Mounts:
      /var/run/cilium from cilium-run (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-swvtx (ro)
```      

Kubectl get pod -o yaml
```
  - args:
    - |-
      for i in {1..5}; do \
        [ -S /var/run/cilium/monitor1_2.sock ] && break || sleep 10;\
        done; \
      cilium monitor --type=trace --type=drop
    command:
    - /bin/bash
    - -c
    - --
    image: quay.io/cilium/cilium:v1.11.3@sha256:cb6aac121e348abd61a692c435a90a6e2ad3f25baa9915346be7b333de8a767f
    imagePullPolicy: IfNotPresent
    name: cilium-monitor
    resources:
      limits:
        memory: 128Mi
      requests:
        cpu: 10m
        memory: 64Mi
    startupProbe:
      failureThreshold: 105
      httpGet:
        host: 127.0.0.1
        httpHeaders:
        - name: brief
          value: "true"
        path: /healthz
        port: 9876
        scheme: HTTP
      periodSeconds: 2
      successThreshold: 1
      timeoutSeconds: 1
```
      
```
NAME           READY   STATUS    RESTARTS   AGE
cilium-j69ht   2/2     Running   0          3m30s
```

As you see all is fine

Reply:
helm diff
<img width="1025" alt="image" src="https://user-images.githubusercontent.com/70581130/163343132-409505e9-b1cc-4bf0-a864-22fdd8ac94ef.png">

Reply:
Perfect, thanks! Looks good!